nixos/modules/services/networking/moodle-dl.nix

```diff
@@ -0,0 +1,63 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.moodle-dl;
+  moodle-dl-json = pkgs.writeText "moodle-dl.json" (builtins.toJSON cfg.config);
+in {
+  options = {
+    services.moodle-dl = {
+      enable = mkEnableOption "moodle-dl, a Moodle downloader";
+
+      config = mkOption {
+        default = {};
+        type = types.attrs;
+        description = ''
+          Configuration for moodle-dl. For a full example, see
+          <link xlink:href="https://github.com/C0D3D3V/Moodle-Downloader-2/wiki/Config.json"/>.
+        '';
+      };
+
+      startAt = mkOption {
+        type = with types; either str (listOf str);
+        description = "When to run moodle-dl. See systemd.time(7) for the format.";
+      };
+
+      directory = mkOption {
+        default = "/var/lib/moodle";
+        type = types.path;
+        description = "The path moodle-dl should download course files to.";
+      };
+
+      package = mkOption {
+        default = pkgs.moodle-dl;
+        type = types.package;
+        description = "The moodle-dl package to use.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.extraUsers.moodle = {
+      isNormalUser = false;
+      home = cfg.directory;
+      createHome = true;
+    };
+
+    system.activationScripts.moodle-dl-config = "ln -sfn ${toString moodle-dl-json} ${cfg.directory}/config.json";
+
+    systemd.services.moodle-dl = {
+      enable = true;
+      serviceConfig = {
+        Type = "oneshot";
+        User = config.users.extraUsers.moodle.name;
+        WorkingDirectory = cfg.directory;
+      };
+      wants = [ "network-online.target" ];
+      script = "${cfg.package}/bin/moodle-dl";
```

Comment:
You can move this to `serviceConfig.ExecStart`.

Reply:
Is there a reason one should use the `serviceConfig` options instead of `script`? If so, do you know why `script` is there in the first place?

Reply:
Either is fine. `script` is just a wrapper around `serviceConfig.ExecStart` for convenience. Specifically `ExecStart` will execute directly and exactly what you pass it, but `script` will create a shell script to wrap what you specify and then execute the shell script.

So if you're just calling a single command it makes more sense to use `ExecStart` to avoid the wrapper which adds no value.

Reply:
Ok, then I'll use the `serviceConfig` ones to avoid the unnecessary wrapping overhead.